app/models/course/assessment.rb

```diff
@@ -131,6 +131,11 @@ def allow_mode_switching?
     submissions.count == 0 || autograded?
   end
 
+  # Check for changes to graded test case booleans for autograded assessments.
+  def regrade_programming_answers?
```

Comment:
This should be a private function. No one else needs to use it, and `previous_changes` isn't useful once the model is saved.